src/data-workspace/print-area/print-area-provider.js

```diff
@@ -0,0 +1,40 @@
+import PropTypes from 'prop-types'
+import React, { useState, useEffect } from 'react'
+import PrintAreaContext from './print-area-context.js'
+
+export default function PrintAreaProvider({ children }) {
+    const [isPrinting, setPrinting] = useState(null)
+    const [isFormCleared, setFormCleared] = useState(null)
+
+    useEffect(() => {
+        if (isPrinting) {
+            window.print()
+        }
```

Comment:
I added a comment - the component is not unmounted, but for effects, the cleanup function is ran on every render so that ensures it's "cleaned" before next run.

<img width="784" alt="image" src="https://user-images.githubusercontent.com/1014725/175011880-fc00c2e2-b0bf-4498-b64e-310c00d7dd47.png">

Reply:
I'm not entirely sure why this works :sweat_smile: but it seems to work